### PR TITLE
Allow more flexibility with repoclosure

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
This is updated based on design discussions. Not much change was required.

File: tests/test_functional.py [hunk 1]
```diff
@@ -185,3 +186,13 @@ def test_obal_add_downstream_hello():
...
 
     assert os.path.islink('packages/hello/hello-2.10.tar.gz')
+
+
+@obal_cli_test(repotype='upstream')
+def test_obal_repoclosure(**kwargs):
+    assert_obal_success(['repoclosure', 'core-repoclosure'])
+
+    expected_log = [
+        "['{bin}/repoclosure', '-c', '{path}/upstream/repoclosure/el7.conf', '-t', '-r', 'scratch']".replace('{path}', kwargs['repodir']),  # noqa: E501
```
Comment (ehelms) [inline tests/test_functional.py]: Basically, I need to know at this point what the generated repo path is and I am not having success getting this value at this point in this Python world. I tried setting a "global" variable but that did not work. This method works expect that it forces a change to the method signature of every other method.

Comment (evgeni) [reply tests/test_functional.py]: I ended up with

```diff
diff --git a/tests/test_functional.py b/tests/test_functional.py
index f03756f..cf6af5f 100644
--- a/tests/test_functional.py
+++ b/tests/test_functional.py
@@ -38,7 +38,6 @@ def obal_cli_test(func=None, repotype='upstream'):
 
         subprocess.check_call(['git', 'init'])
 
-        kwargs['repodir'] = repodir
         func(*args, **kwargs)
 
         os.chdir(oldcwd)
@@ -67,6 +66,7 @@ def assert_obal_failure(args):
 def assert_mockbin_log(content):
     expected_log = "\n".join(content)
     expected_log = expected_log.replace('{bin}', MOCKBIN_DIR)
+    expected_log = expected_log.replace('{pwd}', os.getcwd())
     with open(os.environ['MOCKBIN_LOG']) as mockbinlog:
         log = mockbinlog.read().strip()
         assert log == expected_log
@@ -189,10 +189,10 @@ def test_obal_add_downstream_hello():
 
 
 @obal_cli_test(repotype='upstream')
-def test_obal_repoclosure(**kwargs):
+def test_obal_repoclosure():
     assert_obal_success(['repoclosure', 'core-repoclosure'])
 
     expected_log = [
-        "['{bin}/repoclosure', '-c', '{path}/upstream/repoclosure/el7.conf', '-t', '-r', 'scratch']".replace('{path}', kwargs['repodir']),  # noqa: E501
+        "['{bin}/repoclosure', '-c', '{pwd}/repoclosure/el7.conf', '-t', '-r', 'scratch']",  # noqa: E501
     ]
     assert_mockbin_log(expected_log)
```

**Comment by ehelms** [comment]
Thanks for the patch! Updated

**Pull request merged**